.github/workflows/build-and-push.yaml

```diff
@@ -3,6 +3,7 @@ name: Build & Push Container Image
 # Push a container image when the main branch is pushed (to publish the edge tag) or when a tag is pushed (to publish a
 # release). This isn't dependant on the tests passing or the linter being happy.
 on:
+  pull_request:
```

Comment:
Why add the build step on pull requests?

Reply:
For test building a docker image on a pull request